pkg/utils/test_util.go

```diff
@@ -32,6 +34,19 @@ var (
 
 	// IgnoreConditionLTTAndMessageFields is a cmpopts.IgnoreFields that ignores the LastTransitionTime and Message fields
 	IgnoreConditionLTTAndMessageFields = cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime", "Message")
+
+	MetricsCmpOptions = []cmp.Option{
+		cmpopts.SortSlices(func(a, b *prometheusclientmodel.Metric) bool {
+			return a.GetGauge().GetValue() < b.GetGauge().GetValue() // sort by time
+		}),
+		cmpopts.SortSlices(func(a, b *prometheusclientmodel.LabelPair) bool {
+			return *a.Name < *b.Name // Sort by label name
+		}),
+		cmp.Comparer(func(a, b *prometheusclientmodel.Gauge) bool {
+			return (a.GetValue() > 0) == (b.GetValue() > 0)
```

Comment:
we need to handle nil, otherwise it will panic

for example,
```
cmpopts.SortSlices(func(a, b *prometheusclientmodel.LabelPair) bool {
			if a.Name == nil || b.Name == nil {
                            return a.Name == nil
                        }
                        return *a.Name < *b.Name // Sort by label
		}),
```